tests/transition.md

```diff
@@ -659,3 +662,174 @@ All test cases described in this document are beginning in a pre-Merge world and
+  <details>
+  <summary>Click for details &#9662;</summary>
+
+  * `Genesis <- ... <- INV_TB <- P1`, `INV_TB` is a *valid PoW* but an *invalid terminal* block:
+    * [ ] `INV_TB.TD < TTD`
+    * [ ] `INV_TB.parent.TD >= TTD` -- it might require a second EL with a higher `TTD` value
+  * EL starts with `INV_TB` as the head
+  * `newPayload(P1)`
+    * EL returns `{status: INVALID, latestValidHash: 0x00..00}`
+    * EL's head points to `INV_TB` if `INV_TB.TD < TTD`, and to `INV_TB.parent` if `INV_TB.parent.TD >= TTD`
+  
+  </details>
+
+* [ ] Syncing with the chain having a valid transition,  with enabled `TBH` override
+  <details>
+  <summary>Click for details &#9662;</summary>
+
+  * `A: Genesis <- ... <- Bn <- TB <- P1`, `TB` is a valid terminal block, i.e. `TB.blockHash == TBH`
+  * `B: Genesis <- ... <- Bn`, `B` chain is heavier than `A` and would be canonical unless `TBH` override is activated
+  * EL starts with `B` imported
+  * `newPayload(P1) + forkchoiceUpdated(P1)`
+    * EL returns `{status: SYNCING}`
+    * EL's head points to `Bn`
+  * EL should pull `TB <- P1` from the network
+  * poll `forkchoiceUpdated(P1)`
+    * EL returns `{status: VALID, latestValidHash: null}`
+    * EL's head points to `P1`
+  
+  </details>
+
+* [ ] Stop processing synced PoW blocks with `TBH` override enabled
+  <details>
+  <summary>Click for details &#9662;</summary>
+
+  * PoW client starts with `Genesis <- ... <- TB <- B1 <- B2 <- ... <- Bn` chain, where `TB` is a valid terminal block, i.e. `TB.blockHash == TBH`
+  * EL connects to the PoW client and syncs with the advertised chain
+    * EL's head points to `TB`
+  
+  </details>
+
+### CL client tests
+
+* [ ] Transition on a valid chain with `TBH` override enabled
+  <details>
+  <summary>Click for details &#9662;</summary>
+
+  * `EL.A: Genesis <- ... <- TB`, `TB` is a valid terminal block. i.e. `TB.blockHash == TBH`
+  * `EL.B: Genesis <- ... <- Bn`, `B` chain is heavier than `A` and would be canonical unless `TBH` override is activated
+  * `CL: Genesis <- ... <- Bellatrix`
+  * EL starts with imported `A` and `B`
+  * `TERMINAL_BLOCK_HASH_ACTIVATION_EPOCH` is greater than `BELLATRIX_FORK_EPOCH` 
+  * CL strats with `Genesis` and builds a chain up to `Bellatrix` and upgrades to `Bellatrix`
+    * Merger transition starts only when `TERMINAL_BLOCK_HASH_ACTIVATION_EPOCH` is reached
+  * CL drives EL through transition and finalizes it
+    * The terminal block is `EL.A.TB`, and the PoS chain is built atop of that block
+    * `eth_getBlockByNumber(latest)` returns the head
+    * `eth_getBlockByNumber(safe)` returns the most recent justified block
+    * `eth_getBlockByNumber(finalized)` returns the most recent finalized block
+  
+  </details>
+
+* [ ] Transition before `TERMINAL_BLOCK_HASH_ACTIVATION_EPOCH`
+  <details>
+  <summary>Click for details &#9662;</summary>
+
+  * `EL: Genesis <- ... <- Bn`
+  * Nodes: builder, importer; importer is connected to builder on both layers
```

Comment:
Is in this case the builder a CL+EL combo where the CL has all the validator keys, and the importer a CL+EL combo where the CL has no validator keys ?